sources/host-ctr/cmd/host-ctr/main.go

```diff
@@ -164,106 +164,155 @@ func _main() int {
 		return 0
 	}
 
-	// Clean up target container if it already exists before starting container
-	// task.
-	if err := deleteCtrIfExists(ctx, client, containerID); err != nil {
-		return 1
-	}
-
-	// Get the cgroup path of the systemd service
-	cgroupPath, err := cgroups.GetOwnCgroup("name=systemd")
+	// Check if the target container already exists. If it does, take over the helm to manage it.
+	container, err := client.LoadContainer(ctx, containerID)
```

Comment:
We should also handle toggling superpowered on and off.

Reply:
If the image for the host container is changed via settings. Then the corresponding `host-containers@` service will be restarted by systemd via `restart-commands = ["/usr/bin/host-containers"]` . This means `host-ctr` will actually receive a termination signal and proceed to try and terminate the container task and restart. That hasn't changed.

What this is trying to address is `host-ctr`'s connection to containerd being closed off suddenly and losing track of the container task status and coming back up to try and reclaim the original running host container task.

Reply:
Oh I guess looking at the code, it seems like to actually apply the new host containers setting (whether it be toggling superpowered or changing the image URL) users would have to toggle `enable` and `disable` as well. `host-containers` (the binary doesn't actually handle restarts?

Is that the intended workflow? @zmrow @tjkirch ?

Reply:
It doesn't seem robust to the case where `host-ctr` itself dies for some reason, gets restarted, and finds a running container - which may not be running with the right settings. Previously we did handle this correctly, at the cost of tearing down a running container that was otherwise fine.

Can we inspect the running container and determine that it's correct based on the image and one of the superpowered properties from the spec?

Reply:
You're right, changing the image or toggling superpowered doesn't restart the affected host container today. 😞

Reply:
There might be another edge case to consider - if settings for host-containerd are changed in the same transaction that disables a running host container, then `host-ctr` will exit after losing the connection, and not be restarted afterwards to clean up the running task.

One approach might be for `host-ctr` to have a cleanup mode so that it can be invoked by `host-containers` to delete the running task, if present.

That would handle the edge case and avoid the need to inspect the running container, because it wouldn't be running if the settings had changed.

Reply:
> Is that the intended workflow? @zmrow @tjkirch ?

(Just for historical perspective - it was *intended*, but we didn't *like it* :)  We didn't have the tools at the time to restart properly on those settings changes and considered it a weakness.)